src/components/Swap.tsx

```diff
@@ -322,6 +327,22 @@ const Swap = ({ transferChains }: SwapProps) => {
   // Elements used for animations
   const routeCards = useRef<HTMLDivElement | null>(null)
 
+  useEffect(() => {
+    // get new execution status on page load
+    activeRoutes.map((route) => {
+      if (!web3 || !web3.library) return
+      const settings = {
+        updateCallback: (updatedRoute: RouteType) => {
+          storeRoute(updatedRoute)
+          setActiveRoutes(readActiveRoutes())
+          setHistoricalRoutes(readHistoricalRoutes())
+        },
+      }
+      LiFi.resumeRoute(web3.library.getSigner(), route, settings)
+      LiFi.moveExecutionToBackground(route)
+    })
+  }, [web3.library])
```

Comment:
This will be executed everytime the `web3.library` changes, do we know when this happens? Does this happen on chainswitch or smth like that? 
Is it a problem that it could happen that `resumeRoute` is triggered again and again?

Reply:
Had this thought as well. `web3.library` changes everytime a new library gets loaded. So on page load on connect / disconnect and on change switch. And yes this means, that this sideEffect will be called multiple times. But this is not an issue because:

- Resume Route checks if the route has been halted (moved to bg or stopped). If it has not been halted, it will not execute.
  - Triggering an execution multiple times does no harm
- Triggering this method after a chain switch is a actually good thing:
   - If the step only has one route, nothing really happens. The route will be marked as DONE if done and moved to historical TXs (assuming all txs have been sent)
   - If the routes has multiple steps: the retrigger will alert the user that a chain switch is required, as the wrong signer for the tx is passed. This is a nice clue, that the execution is not execution in the background anymore and the reason for it.